packages/shared/src/components/filters/ContentTypesFilter.tsx

```diff
@@ -1,40 +1,172 @@
 import React, { ReactElement } from 'react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
 import { FilterCheckbox } from '../fields/FilterCheckbox';
 import { useAdvancedSettings } from '../../hooks/feed';
 import useFeedSettings from '../../hooks/useFeedSettings';
 import { getVideoSetting } from './helpers';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../typography/Typography';
+import { Divider, SharedFeedPage } from '../utilities';
+import { LanguageDropdown } from '../profile/LanguageDropdown';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { ContentLanguage } from '../../lib/user';
+import { UPDATE_USER_PROFILE_MUTATION } from '../../graphql/users';
+import { gqlClient } from '../../graphql/common';
+import { useToastNotification } from '../../hooks';
+import { labels } from '../../lib';
+import { useFeature } from '../GrowthBookProvider';
+import { feature } from '../../lib/featureManagement';
+import { OtherFeedPage, RequestKey } from '../../lib/query';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent, TargetType } from '../../lib/log';
 
 export function ContentTypesFilter(): ReactElement {
+  const postTitleLanguageFeature = useFeature(feature.postTitleLanguage);
+  const queryClient = useQueryClient();
   const { advancedSettings, isLoading } = useFeedSettings();
   const { selectedSettings, onToggleSettings } = useAdvancedSettings();
+  const { user, updateUser } = useAuthContext();
+  const { displayToast } = useToastNotification();
+  const { logEvent } = useLogContext();
 
   const videoSetting = getVideoSetting(advancedSettings);
 
+  const { mutate: onLanguageChange } = useMutation(
+    async (value?: ContentLanguage) => {
+      await updateUser({
+        ...user,
+        language: value,
+      });
+
+      await gqlClient.request(UPDATE_USER_PROFILE_MUTATION, {
+        data: {
+          language: value,
+        },
+      });
+    },
+    {
+      onMutate: (value) => {
+        logEvent({
+          event_name: LogEvent.ChangeSettings,
+          target_type: TargetType.Language,
+          target_id: value,
+        });
+      },
+      onSuccess: async () => {
+        const requestKeys = [
+          ...Object.values(SharedFeedPage),
+          OtherFeedPage.Preview,
+          RequestKey.Squad,
+          RequestKey.FeedPreview,
+          RequestKey.FeedPreviewCustom,
+          RequestKey.PostKey,
+          RequestKey.Bookmarks,
+          RequestKey.ReadingHistory,
+          RequestKey.RelatedPosts,
+          RequestKey.SourceFeed,
+          RequestKey.SourceMostUpvoted,
+          RequestKey.SourceBestDiscussed,
+          RequestKey.TagFeed,
+          RequestKey.TagsMostUpvoted,
+          RequestKey.TagsBestDiscussed,
+        ];
```

Comment:
Instead of changing query keys all over the place I opted into invalidating data that contains posts. This will mean that when user again goes to some feed or post the data will simply refetch then and show new translated titles if exist.

Reply:
Hmm don't we prefix them with feed or something?
That would have been smart 😂

Reply:
Bad thing here is that if user changes the language in other tab it will not invalidate but for experiment I think its ok. If this wins and we want to add more translations eg. to content we have to introduce `lang` to query keys so they are invalidated automatically.

Reply:
> Hmm don't we prefix them with feed or something?
That would have been smart 😂

I though we did and then I seen https://github.com/dailydotdev/apps/pull/3469#discussion_r1725077628